lib/core/test_suite.js

```diff
@@ -208,6 +208,26 @@ export class TestSuite {
 
   #evaluateSuiteDefinition() {
     this.#body.call();
+    this.#evaluateExclusiveRunsMarks();
+  }
+
+  #evaluateExclusiveRunsMarks() {
+    const testsMarkedAsOnlyCount = this.#getTestsMarkedAsExclusive();
+    if (testsMarkedAsOnlyCount > 0) {
+      this.#skipTestsNotMarkedAsExclusive();
+    }
+  }
+
+  #getTestsMarkedAsExclusive() {
+    return this.tests().filter(test => test.isMarkedAsExclusiveForRun()).length;
+  }
```

Comment:
we can gain more expressiveness by using `some`:

```suggestion
  #evaluateExclusiveRunsMarks() {
    if (this.#thereAreTestsMarkedAsExclusive()) {
      this.#skipTestsNotMarkedAsExclusive();
    }
  }

  #thereAreTestsMarkedAsExclusive() {
    return this.tests().some(test => test.isMarkedAsExclusiveForRun());
  }
```